addons/mail/models/mail_message.py

```diff
@@ -737,7 +737,9 @@ def unlink(self):
         messages_by_partner = defaultdict(lambda: self.env['mail.message'])
         partners_with_user = self.partner_ids.filtered('user_ids')
         for elem in self:
-            for partner in elem.partner_ids & partners_with_user:
+            for partner in (
+                elem.partner_ids & partners_with_user | elem.notification_ids.author_id
+            ):
```

Comment:
What is the link with unlinked notifications issue ?

Reply:
For now only recipients are notified to remove new message notifications from their messaging menu. I added the author of the message as well in case a failure notification is displayed to remove it from the messaging menu as well.